### PR TITLE
[path_provider] Move package into a path_provider directory

### PR DESCRIPTION
**Pull request by franciscojma86**
## Description

First step for federation, as outlined in https://github.com/flutter/flutter/issues/46307#issuecomment-589301237

## Related Issues

https://github.com/flutter/flutter/issues/46307

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
@franciscojma86 can you edit the PR description to describe what "First step of federation" is, and how many steps there will be in total?

File: packages/path_provider/path_provider/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: path_provider
 description: Flutter plugin for getting commonly used locations on the Android &
   iOS file systems, such as the temp and app data directories.
 homepage: https://github.com/flutter/plugins/tree/master/packages/path_provider
```
Comment (ditman) [inline packages/path_provider/path_provider/pubspec.yaml]: The homepage also changes, to `https://github.com/flutter/plugins/tree/master/packages/path_provider/path_provider`

Comment (ditman) [reply packages/path_provider/path_provider/pubspec.yaml]: (you can push this minor change when you come back to this file when doing the federation of _macos, just to conserve your CI/CD luck ;) )

Comment (franciscojma86) [reply packages/path_provider/path_provider/pubspec.yaml]: Did here :)

**Pull request merged**